modules/blocks/verify.js

```diff
@@ -319,27 +319,29 @@ __private.verifyPayload = function(block, result) {
 	const appliedTransactions = {};
 
 	for (const i in block.transactions) {
-		const transaction = block.transactions[i];
-		let bytes;
-
-		try {
-			bytes = library.logic.transaction.getBytes(transaction);
-		} catch (e) {
-			result.errors.push(e.toString());
-		}
+		if (Object.prototype.hasOwnProperty.call(block.transactions, i)) {
```

Comment:
`block.transactions` is an array, I don't think this check is necessary here and introduces confusion in the critical function - `blocks.verify`.

Reply:
Overall this rule looks odd to me. We should either introduce a rule of not-using `for ... in` at all, or keep this flag off.

Reply:
This code will change when introducing `no-restricted-syntax` https://github.com/LiskHQ/lisk/pull/2643 . Perhaps it'll make more sense after it.